templates/upgrade.j2

```diff
@@ -1,145 +1,76 @@
+### - $domain
+### - $path
+### - $language
+### - $install_dir
+### - $port
+### ...
+
+### In the context of upgrade,
+### - resources are automatically provisioned / updated / deleted (depending on existing resources)
+### - a safety backup is automatically created by the core and will be restored if the upgrade fails
+{% endif %}
 
-#=================================================
-# STANDARD UPGRADE STEPS
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
 {% if data.generator_mode == 'tutorial' -%}
-#ynh_script_progression --message="Ensuring downward compatibility..."
+#ynh_script_progression "Ensuring downward compatibility..."
 
-#
-# N.B. : the followings setting migrations snippets are provided as *EXAMPLES*
-# of what you may want to do in some cases (e.g. a setting was not defined on
-# some legacy installs and you therefore want to initialize stuff during upgrade)
-#
+### N.B. : the following setting migration snippets are provided as *EXAMPLES*
+### of what you may want to do in some cases (e.g. a setting was not defined on
+### some legacy installs and you therefore want to initiaze stuff during upgrade)
 
 # If db_name doesn't exist, create it
-#if [ -z "${db_name:-}" ]; then
-#	db_name=$(ynh_sanitize_dbid --db_name=$app)
-#	ynh_app_setting_set --app=$app --key=db_name --value=$db_name
-#fi
+# ynh_app_setting_set_default --key=db_name --value="$(ynh_sanitize_dbid --db_name=$app)"
 
 # If install_dir doesn't exist, create it
-#if [ -z "${install_dir:-}" ]; then
-#	install_dir=/var/www/$app
-#	ynh_app_setting_set --app=$app --key=install_dir --value=$install_dir
-#fi
+# ynh_app_setting_set_default --key=install_dir --value="/var/www/$app"
 {% endif %}
 
 {% if data.main_technology not in ["php", "none"] -%}
 #=================================================
 # STOP SYSTEMD SERVICE
 #=================================================
-ynh_script_progression --message="Stopping a systemd service..."
+ynh_script_progression "Stopping a systemd service..."
 
-ynh_systemd_action --service_name=$app --action="stop" --log_path="/var/log/$app/$app.log"
+ynh_systemctl --service="$app" --action="stop"
 {% endif %}
 
-#=================================================
-# "REBUILD" THE APP (DEPLOY NEW SOURCES, RERUN NPM BUILD...)
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
+ynh_script_progression --message="Upgrading source files..."
 
-if [ "$upgrade_type" == "UPGRADE_APP" ]
-then
-	ynh_script_progression --message="Upgrading source files..."
-
-	# Download, check integrity, uncompress and patch the source from app.src
-	ynh_setup_source --dest_dir="$install_dir" --full_replace=1  --keep=".env"
-	{{ data.custom_config_file }}
-fi
+ynh_setup_source --dest_dir="$install_dir" --full_replace --keep=""
```

Comment:
That empty "keep" is a bit confusing here ... either we should remove it or add a comment about the fact that it should be filled with a list of files/folders that need to be kept when replacing the intall dir

Reply:
what about removing `--full_replace` and `--keep=""` and adding a link to the helpers doc? 
```
{% if data.generator_mode == 'tutorial' -%}
# https://yunohost.org/en/packaging_apps_helpers_v2.1#ynh-setup-source
{% endif %}
```

Reply:
Yeah I guess we can do this though Sala mandar was pushing for --full-replace to become the standard ... but yeah as a middleground I would go for adding a comment in all cases (not just the tutorial mode) to encourage people to read about --full-replace and --keep : 

```suggestion
# Please note that you are encouraged to use --full_replace here,
# in conjunction with --keep="..." where you should list config file
# and other file/folders to be kept when replacing the install dir.
# cf. https://yunohost.org/en/packaging_apps_helpers_v2.1#ynh-setup-source
ynh_setup_source --dest_dir="$install_dir"
```

Reply:
wait... the [confil file](https://github.com/YunoHost/appgenerator/blob/50d4a3d87d73ffb12bd42104242c1412f4ba2b2a/templates/upgrade.j2#L89) was already declared but not in the right place...